python-squid/src/main/java/org/sonar/python/api/PythonGrammar.java

```diff
@@ -245,15 +245,17 @@ public static void grammar(LexerfulGrammarBuilder b) {
     b.rule(FPLIST).is(FPDEF, b.zeroOrMore(",", FPDEF), b.optional(","));
 
     b.rule(TYPEDARGSLIST).is(b.firstOf(
-      b.sequence(b.zeroOrMore(TFPDEF, b.optional("=", TEST), ","), b.firstOf(b.sequence("*", NAME, b.optional(",", "**", NAME)), b.sequence("**", NAME))),
-      b.sequence(TFPDEF, b.optional("=", TEST), b.zeroOrMore(",", TFPDEF, b.optional("=", TEST)), b.optional(",")))
-    );
+      b.sequence("**", TFPDEF),
+      b.sequence("*", b.optional(TFPDEF), b.zeroOrMore(",", TFPDEF, b.optional("=", TEST)), b.optional(",", "**", TFPDEF)),
+      b.sequence(TFPDEF, b.optional("=", TEST), b.zeroOrMore(",", TFPDEF, b.optional("=", TEST)), b.optional(","), b.optional(b.firstOf(
```

Comment:
@vilchik-elena It's strange to have `b.optional(",")`. As it is, we match `p *p`.